### PR TITLE
Unified About: Add Tracking

### LINKED ISSUE #17416: [iOS] About Screen v1: Add tracks

### PR DESCRIPTION
**Pull request by diegoreymendez**
Initial PR for #17416.

Implements tracking for showing the about dialog, dismissing it, and pressing on any button within that screen.  It also covers the event tracking for "Legal and More".

### Test 1:

- Go to Me > About
- Verify that you see:

🔵 Tracked: about_screen_shown <screen: main>

- Tap on any menu item and verify you see:

🔵 Tracked: about_screen_button_tapped <button: $button_name>

- Close the dialog, and verify you see:

🔵 Tracked: about_screen_dismissed <screen: main>

### Test 2: Test tapping buttons

Test tapping the buttons and make sure you get:

🔵 Tracked: about_screen_button_tapped <button: $button_name>

### Test 3: Legal and more

- Test tapping on "Legal And More"

🔵 Tracked: about_screen_shown <screen: legal_and_more>

- Tap on any menu item and verify you see:

🔵 Tracked: about_screen_button_tapped <button: $button_name>

- Dismiss the window by pressing back, tapping done or sliding it down.

🔵 Tracked: about_screen_dismissed <screen: legal_and_more>

## Regression Notes

1. Potential unintended areas of impact
2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Me/App Settings/About/UnifiedAboutViewController.swift [hunk 1]
```diff
@@ -6,12 +6,14 @@ struct AboutItem {
     let title: String
     let subtitle: String?
     let cellStyle: AboutItemCellStyle
+    let eventButton: UnifiedAboutEvent.Button?
     let action: (() -> Void)?
 
-    init(title: String, subtitle: String? = nil, cellStyle: AboutItemCellStyle = .default, action: (() -> Void)? = nil) {
+    init(title: String, subtitle: String? = nil, cellStyle: AboutItemCellStyle = .default, eventButton: UnifiedAboutEvent.Button?, action: (() -> Void)? = nil) {
```
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/Me/App Settings/About/UnifiedAboutViewController.swift]: I added the `eventButton` parameter without a default value on purpose.

My rationale is that if a new entry is added to this the table view, I want to make sure we don't forget to add the corresponding event button.  We will have to explicitly set it to `nil` if there's nothing to track.

**Review by frosty — COMMENTED** [review]
Overall this is looking good, and tracking is working as expected 👍 Just a few general notes:

* It seems as though we won't actually be doing anything when the user interacts with the specific app icons (just the Family heading), so we can remove the tracking for those.
* I was thinking about how we'll make this extendible for other apps using the screen, when they may have different options or titles. I think the main issue there is that the different button events are all hardcoded into the Button enum. Perhaps we could just pass a tracking identifer (string) into the AboutItem initializer instead? 🤔 (I had wondered if we could just snake_case the button title, but that wouldn't be useful if the title was localized) What do you think?

**Comment by mokagio** [comment]
@diegoreymendez  I'm going to bump this to the next release because we'll be code freezing 18.7 today and this hasn't been approved yet.

If this cannot wait two weeks and it's important that it makes it into this release, let me know and we'll organize a new beta once ready.

**Comment by diegoreymendez** [comment]
@frosty - Ready for another look.  Since the tracker will be part of the App now, I think we should be fine with using enums.

File: WordPress/Classes/ViewRelated/Me/App Settings/About/AboutScreenConfiguration.swift [hunk 2]
```diff
@@ -9,6 +9,14 @@ protocol AboutScreenConfiguration {
 
     /// A block that dismisses the about screen
     var dismissBlock: ((AboutItemActionContext) -> Void) { get }
+
+    /// Called when the screen will be shown for customization purposes and event tracking.
+    ///
+    func willShow(viewController: UIViewController)
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Me/App Settings/About/AboutScreenConfiguration.swift:15]: Should we either make these into blocks, or change dismissBlock to a method, for consistency?

**Comment by mokagio** [comment]
@diegoreymendez, in case you missed it after opening the PR because of slow CI time, there are [some build errors in the tests](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-iOS/25732/workflows/fd463bba-6b06-4c3c-a706-8b36d1eabd8f/jobs/60737?invite=true#step-112-5542). Examples:

```
▸ Compiling QueueTests.swift

❌  /Users/distiller/project/WordPress/WordPressTest/Me/App Settings/About/UnifiedAboutEventTests.swift:4:23: cannot find type 'UnifiedAboutEvent' in scope

fileprivate extension UnifiedAboutEvent.Button {
                                                                                                                                                                                  ^~~~~~~~~~~~



❌  /Users/distiller/project/WordPress/WordPressTest/Me/App Settings/About/UnifiedAboutEventTests.swift:45:20: cannot find type 'UnifiedAboutEvent' in scope

        let event: UnifiedAboutEvent = .screenShown
                      ^~~~~~~~~~~~~~~~~
```

**Review by frosty — COMMENTED** [review]
I've run through all the tests, all looks good. Just one minor code comment / query.

I also noticed that the Share button actually also triggers another existing analytics event, although I guess it's not a problem tracking this new one as well: `recommend_app_engaged <source: about>`.

**Comment by Gio2018** [comment]
@diegoreymendez @frosty I gave this another pass based on the comments, please have another look, thank you!!

**Review by frosty — COMMENTED** [review]
Changes are looking good! But tests are failing as there's been quite a bit of change in the design of this since they were written.

**Review by frosty — APPROVED** [review]
Let's :shipit:!

**Pull request merged**